src/Common/src/Interop/Unix/System.Native/Interop.ReadDir.cs

```diff
@@ -12,7 +12,19 @@ internal static partial class Interop
 {
     internal static partial class Sys
     {
-        internal static int ReadBufferSize { get; } = GetReadDirRBufferSize();
+        internal static object s_syncObject = new object();
```

Comment:
This needs to go upstream first. You replaced explicit static ctor but I think we can do without s_syncObject